rewrite-java/src/main/java/org/openrewrite/java/MethodMatcher.java

```diff
@@ -195,7 +195,7 @@ boolean matchesTargetType(@Nullable JavaType.FullyQualified type) {
 
         if (targetTypePattern.matcher(type.getFullyQualifiedName()).matches()) {
             return true;
-        } else if (!type.getFullyQualifiedName().equals("java.lang.Object") && matchesTargetType(type.getSupertype() == null ? JavaType.Class.build("java.lang.Object") : type.getSupertype())) {
+        } else if (!type.getFullyQualifiedName().equals("java.lang.Object") && matchesTargetType(JavaType.Class.build("java.lang.Object"))) {
```

Comment:
This "else if" clause should be removed because we check the super type (if the `matchOverrides` is `'true`) right after this. 

it probably should be something like : 

```
    boolean matchesTargetType(@Nullable JavaType.FullyQualified type) {
        if (type == null) {
            return false;
        }

        if (targetTypePattern.matcher(type.getFullyQualifiedName()).matches()) {
            return true;
        }

        if (matchOverrides) {
            JavaType.FullyQualified superType = type.getSupertype() == null ? JavaType.Class.build("java.lang.Object") : type.getSupertype(); 
            if (matchesTargetType(superType)) {
                return true;
            }

            for (JavaType.FullyQualified anInterface : type.getInterfaces()) {
                if (matchesTargetType(anInterface)) {
                    return true;
                }
            }
        }
        return false;
    }
```

Reply:
Taking a look at this now-- we do need to include the check which asks, "do you have a supertype, AND are you (not) currently java.lang.Object?", because otherwise it'll throw up a stackoverflow, infinitely saying "if your supertype is null, try against the supertype of java.lang.Object". 

But "java.lang.Object"'s supertype is null. So it'll call `matchesTargetType` using "java.lang.Object". Which has a null supertype. So it'll call `matchesTargetType` using "java.lang.Object". Which has a null supertype. So it'll call `matchesTargetType` using "java.lang.Object". Which -- (😉 )

Making a fix for it now

Reply:
fair enough, we can add a short circuit for this: 

```
if (matchOverrides && !type.getFullyQualifiedName().equals("java.lang.Object")) {
```